core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -61,6 +61,10 @@ private DateTimeUtils() {}
 
     public static int parseDate(String value)
     {
+        // in order to follow the standard, we should validate the value:
+        // - require format 'YYYY-MM-DD'
+        // - only allow unsigned numbers
```

Comment:
`unsigned numbers` - does this talk about each component (year, month and date) or does it talk about the date as a whole?

Reply:
The spec says that each component shall be unsigned (year, month, day). I'll update the comment to make it clearer.

Reply:
> If so then how can negative dates be represented?

Seems like they just can't. 

I'm only reporting an issue I found when fixing coercions. I don't think we will decide to remove the support for negative dates.